src/BenchmarkDotNet/Locators/ProjectLocator.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using BenchmarkDotNet.Exporters;
+
+namespace BenchmarkDotNet.Locators;
+
+public class ProjectLocator : ILocator
+{
+    public static ProjectLocator Default { get; } = new ProjectLocator();
+
+    public LocatorType LocatorType => LocatorType.ProjectFile;
+
+    public FileInfo Locate(DirectoryInfo rootDirectory, Type type)
+    {
+        // important assumption! project's file name === output dll name
+        string projectName = type.GetTypeInfo().Assembly.GetName().Name;
+
+        var possibleNames = new HashSet<string> { $"{projectName}.csproj", $"{projectName}.fsproj", $"{projectName}.vbproj" };
+        var projectFiles = rootDirectory
+                           .EnumerateFiles("*proj", SearchOption.AllDirectories)
+                           .Where(file => possibleNames.Contains(file.Name))
+                           .ToArray();
+
+        if (projectFiles.Length == 0)
+        {
+            throw new NotSupportedException(
```

Comment:
If we're now supporting multiple locators, this should not throw. Perhaps the ILocator should use `bool TryLocate(..., out string errorMessage)` instead.

Reply:
Is `TryLocate()` or `FileInfo? Locate()` preferred in BenchmarkDotNet? I'll make it into a TryLocate for now.

Reply:
I think it depends on the usage. Some places use `??` to get fallback values, but since this is a public API, I think `Try*` makes more sense.